frontend/src/container/GridGraphLayout/Graph/FullView/GraphManager.tsx

```diff
@@ -0,0 +1,215 @@
+import { memo, useEffect, useState } from 'react';
+
+import { getGraphVisiblityStateOnDataChange } from '../utils';
+import CheckBox from './CheckBox';
+import { ColumnsTitle, DataIndexAndKey } from './contants';
+import {
+	FilterTableAndSaveContainer,
+	FilterTableContainer,
+	LabelContainer,
+	SaveContainer,
+} from './styles';
+import { DataSetProps, ExtendedChartDataset, LegendEntryProps } from './types';
+import {
+	getAbbreviatedLabel,
+	getDefaultTableDataSet,
+	saveLegendEntriesToLocalStorage,
+} from './utils';
+
+function GraphManager({
+	data,
+	graphVisibilityStateHandler,
+	name,
+}: GraphManagerProps): JSX.Element {
+	const [graphVisibilityState, setGraphVisibilityState] = useState<boolean[]>(
+		Array(data.datasets.length).fill(true),
+	);
+	const [tableDataSet, setTableDataSet] = useState<ExtendedChartDataset[]>(
+		getDefaultTableDataSet(data),
+	);
+	const [legendEntries, setLegendEntries] = useState<LegendEntryProps[]>([]);
+
+	const { notifications } = useNotifications();
+
+	useEffect(() => {
+		if (graphVisibilityStateHandler) {
+			graphVisibilityStateHandler([...graphVisibilityState]);
+		}
+	}, [graphVisibilityState, graphVisibilityStateHandler]);
+
+	useEffect(() => {
+		const newGraphVisibilityStates: boolean[] = Array(data.datasets.length).fill(
+			true,
+		);
+		setTableDataSet(getDefaultTableDataSet(data));
+		data.datasets.forEach((dataset, i) => {
+			const index = legendEntries.findIndex(
+				(legendEntry) => legendEntry.label === dataset.label,
+			);
+			if (index !== -1) {
+				newGraphVisibilityStates[i] = legendEntries[index].show;
+			}
+		});
+		setGraphVisibilityState(newGraphVisibilityStates);
+	}, [data, legendEntries]);
+
+	useEffect(() => {
+		const visibilityStateAndLegendEntry = getGraphVisiblityStateOnDataChange(
+			data,
+			false,
+			name,
+		);
+		setGraphVisibilityState(visibilityStateAndLegendEntry.graphVisibilityStates);
+		setLegendEntries(visibilityStateAndLegendEntry.legendEntry);
+	}, [data, name]);
+
+	const checkBoxOnChangeHandler = (
+		e: CheckboxChangeEvent,
+		index: number,
+	): void => {
+		graphVisibilityState[index] = e.target.checked;
+		setGraphVisibilityState([...graphVisibilityState]);
+		if (graphVisibilityStateHandler) {
+			graphVisibilityStateHandler(graphVisibilityState);
+		}
+	};
+
+	const getCheckBox = (index: number): React.ReactElement => (
+		<CheckBox
+			data={data}
+			index={index}
+			graphVisibilityState={graphVisibilityState}
+			checkBoxOnChangeHandler={checkBoxOnChangeHandler}
+		/>
+	);
+
+	const labelClickedHandler = (labelIndex: number): void => {
+		const newGraphVisibilityStates = Array(data.datasets.length).fill(false);
+		newGraphVisibilityStates[labelIndex] = true;
+		setGraphVisibilityState([...newGraphVisibilityStates]);
+		if (graphVisibilityStateHandler) {
+			graphVisibilityStateHandler(newGraphVisibilityStates);
+		}
+	};
+
+	const getLabel = (label: string, labelIndex: number): React.ReactElement => (
```

Comment:
can we make a separate component?

Reply:
Okay

Reply:
if there is a separate component then i don't think we need this handler same for other as those are passing the data

Reply:
also we can directly mapping the component to the render to avoid the component

Reply:
Okay got it.